Romiguier/Voleurs/src/Coffre.ts

```diff
@@ -0,0 +1,56 @@
+import type { EtatCoffre } from "./EtatCoffre";
+
+type CoffreParams = {
+  id: number;
+  nombreDePositions: number;
+  nombreDeSymboles: number;
+};
+
+export class Coffre {
+  private _id: number;
+  private _nombreDeCombinaisonsRestantes: number = 0;
+  private _etat: EtatCoffre = "fermé";
+  constructor({ id, nombreDePositions, nombreDeSymboles }: CoffreParams) {
+    this._nombreDeCombinaisonsRestantes = Math.pow(
+      nombreDeSymboles,
+      nombreDePositions
+    );
+    this._id = id;
+  }
+
+  get id() {
+    return this._id;
+  }
+
+  get nombreDeCombinaisonsRestantes(): number {
+    return this._nombreDeCombinaisonsRestantes;
+  }
+
+  get estOuvert(): boolean {
+    return this._etat === "ouvert";
+  }
+
+  get estFerme(): boolean {
+    return this._etat === "fermé";
+  }
+
+  get estEnCoursDOuverture(): boolean {
```

Comment:
Does a Coffre needs to know it is "en cours d'ouverture" ? For me a Coffree is open or close and it also knows its combination, that's it. If someone is working to hack it, that should be handle elsewhere (BandeDeVoleurs look like a good possibility).

Reply:
Done. I have removed all unecessary states not needed without a time loop.